qutebrowser/config/config.py

```diff
@@ -640,23 +640,32 @@ class StyleSheetObserver(QObject):
     Attributes:
         _obj: The object to observe.
         _stylesheet: The stylesheet template to use.
+        _options: The config options that the stylesheet uses. When it's not
+                  necessary to listen for config changes, this attribute may be
+                  missing.
     """
 
     def __init__(self, obj: QObject,
-                 stylesheet: typing.Optional[str],
-                 update: bool) -> None:
+                 stylesheet: Optional[str], update: bool) -> None:
         super().__init__()
         self._obj = obj
         self._update = update
 
         # We only need to hang around if we are asked to update.
-        if self._update:
+        if update:
             self.setParent(self._obj)
         if stylesheet is None:
             self._stylesheet = obj.STYLESHEET  # type: str
         else:
             self._stylesheet = stylesheet
 
+        if update:
+            self._options = jinja.template_config_variables(
+                self._stylesheet)  # type: FrozenSet[str]
```

Comment:
Is the `type:` annotation needed?

Reply:
Not for understanding code correctness, but that reads like "checking membership for this container is fast" to me.